src/main/java/com/parsing/parsers/pdf/parsing/PageOCRPreparator.java

```diff
@@ -44,21 +45,25 @@ public class PageOCRPreparator {
 
     public String preparePage(PDDocument document, int page) throws IOException {
         log.debug("Class PageOCRPreparator started.");
-        String fileResult = "preparePage" + page + ".png";
-        String pagePDF = getPagePDF(document, page);
-        String rotatedPage = rotationImage.rotate(pagePDF);
-        Rectangle pageRectangle = getPageMainRectangle(document, page);
-        Mat tableMat = Imgcodecs.imread(rotatedPage);
-        Mat matPage = cleanTableBorders(tableMat, pageRectangle);
-        Imgcodecs.imwrite(page + ".png", matPage);
-        java.util.List<Rectangle> pageRectanglesAllWords = getPageRectanglesAllWords(page + ".png");
-        List<Rectangle> rectanglesWithSymbols = extractTextFromRectangle(page + ".png", pageRectanglesAllWords);
-        saveRectanglesOnImage(rectanglesWithSymbols, page + ".png");
-        final Rectangle mainPageRectangle = findMainPageRectangle(rectanglesWithSymbols);
-        final Mat result = cleanTableBorders(matPage, mainPageRectangle);
-        Imgcodecs.imwrite(fileResult, result);
-        log.debug("Class PageOCRPreparator finished.");
-        return fileResult;
+        try {
+            String fileResult = "preparePage" + page + ".png";
+            String pagePDF = getPagePDF(document, page);
+            String rotatedPage = rotationImage.rotate(pagePDF);
+            Rectangle pageRectangle = getPageMainRectangle(document, page);
+            Mat tableMat = Imgcodecs.imread(rotatedPage);
+            Mat matPage = cleanTableBorders(tableMat, pageRectangle);
+            Imgcodecs.imwrite(page + ".png", matPage);
+            List<Rectangle> pageRectanglesAllWords = getPageRectanglesAllWords(page + ".png");
+            List<Rectangle> rectanglesWithSymbols = extractTextFromRectangle(page + ".png", pageRectanglesAllWords);
+            saveRectanglesOnImage(rectanglesWithSymbols, page + ".png");
+            final Rectangle mainPageRectangle = findMainPageRectangle(rectanglesWithSymbols);
+            final Mat result = cleanTableBorders(matPage, mainPageRectangle);
+            Imgcodecs.imwrite(fileResult, result);
+            log.debug("Class PageOCRPreparator finished.");
+            return fileResult;
+        } catch (IOException e) {
+            throw new RotationImageException("Rotation image failed", e);
+        }
```

Comment:
If we have IO Exception add please finally block with close method, google it with Java IO finally block or something

Reply:
done